### PR TITLE
Fix route_to issue

### LINKED ISSUE #267: Links created with route_to() do not work when baseURL has subfolders
Let's say `baseURL` is `https://mydomain.com/subfolder`.
Then `route_to('login')` is `/login`.

But it should be `/subfolder/login`.

I suggest to change all `route_to(...)` to `site_url(route_to(...))`.

### PR DESCRIPTION
**Pull request by parisiam**
Fixes #267

- Replaced `route_to()` by `url_to()` in the views, when is was used alone.
- When `route_to()` was used with `site_url()`, as in `site_url(route_to(...))`, I've left it for the sake of diversity.
- Changed `route_to()` to `url_to()` also in `tests/Authentication/MagicLinkTest.php` (please check carefully for this one as I'm not too sure about it).
> First PR here, sorry in advance if I missed something. Remarks/advice welcome.

### CODE REVIEW

File: tests/Authentication/MagicLinkTest.php [hunk 1]
```diff
@@ -37,29 +37,29 @@ protected function setUp(): void
 
     public function testCanSeeMagicLinkForm(): void
     {
-        $result = $this->get(route_to('magic-link'));
+        $result = $this->get(url_to('magic-link'));
```
Comment (kenjis) [inline tests/Authentication/MagicLinkTest.php:40]: Revert all the changes in this file. `$this->get()`, `$this->post()`, `$result->assertRedirectTo()` does not take absolute URL.

See
- https://codeigniter4.github.io/CodeIgniter4/testing/feature.html#requesting-a-page
- https://codeigniter4.github.io/CodeIgniter4/testing/response.html#assertredirectto

**Comment by MGatner** [comment]
Thanks @parisiam! Glad to have your contribution. Please revert the test case changes as kenjis noted. 

> When `route_to()` was used with `site_url()` ... I've left it for the sake of diversity.

I would be in favor of consistency here, open to other opinions though.

**Comment by kenjis** [comment]
I think we don't need to use  `site_url(route_to(...))`.
We use it just because we did not know `url_to()` works like that.

**Comment by parisiam** [comment]
Thank you both for your comments and your help !! Really appreciate it.
- I've reverted the changes in MagicLinkTest.php
- I've replaced site_url(route_to(...)) by url_to(...) in email_2fa_show.php and email_2fa_verify.php
Don't hesitate to let me know if some further corrections are needed.

**Review by kenjis — APPROVED** [review]
Thank you.
LGTM!

**Review by MGatner — APPROVED** [review]
Very nice work! Thank you @parisiam

**Pull request merged**